packages/server/src/schema/uuid/abstract-entity/resolvers.ts

```diff
@@ -34,4 +49,45 @@ export const resolvers: InterfaceResolvers<'AbstractEntity'> &
       return entityRevision.__typename
     },
   },
+  EntityMutation: {
+    async checkoutRevision(_parent, { input }, { dataSources, userId }) {
+      assertUserIsAuthenticated(userId)
+
+      const revision = await dataSources.model.serlo.getUuid({
+        id: input.revisionId,
+      })
+
+      if (revision === null || !isEntityRevision(revision)) {
+        throw new UserInputError('revisionId must belong to a revision')
+      }
+
+      const scope = await fetchScopeOfUuid({
+        id: revision.repositoryId,
+        dataSources,
+      })
+      await assertUserIsAuthorized({
+        userId,
+        dataSources,
+        message: 'you are not authorized to checkout a revision',
```

Comment:
Please be consistent with the existing messages, i.e.
```suggestion
        message: 'You are not allowed to check out the provided revision.',
```